### PR TITLE
Feature calendar 7

### PR DESCRIPTION
**Pull request by hapalacios**
### What this PR does (required): FE: Calendar #7
- PR functionality: this PR enables the Calendar features.
- Calendar page and components added.
- Clicking on a card (event) display their details.
- Cards can be dragged to different dates.

### Screenshots / Videos (required):
- Calendar:
Screen Shot 2021-08-12 at 1 58 47 AM

### Any information needed to test this feature (required):
- To test this feature click on the event to see details or move the event from one day to the other with righ-click button of the mouse pressed.
- Data is static. You can move the event but not make changes yet.

https://user-images.githubusercontent.com/60803185/129147124-22257028-456b-4015-aad2-706551c7b18d.mp4

### Any issues with the current functionality (optional):
- This PR has a better styling than other PR. I will recommend to use this stylying when merging conflicts.
- Protected Route, Calendar.tsx, Avatar and Dashboard.tsx components/Pages were updated. Protected Route is the one validating user. Is not necesary to validate user in each page or component.

### CODE REVIEW

**Review by ajaytomgeorge — COMMENTED** [review]
@hapalacios  Good work :-) . Some comments though,

- Initial commit messages can be made better, if all of them looks similar its hard for the reviewer
- Fix line sequencing issues, so that files like user.js doesn't come as fully changed
- If you are not using the code, please remove it, don't keep it as comment, as it would affect the code quality

File: client/src/components/CalendarBoard/CalendarBoard.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    console.log('current target');
+  };
+
+  const showEventDoubleClick = (): void => {
+    //console.log('Event after double click');
+  };
+
+  const showEventSelected = (event: any): void => {
+    //console.log('Here show event details after event selection');
+    //console.log(event);
```
[the remark this answers is not in the document]
Comment (hapalacios) [reply client/src/components/CalendarBoard/CalendarBoard.tsx]: Done. Thank you.

File: client/src/components/CalendarBoard/CalendarBoard.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    setEvents(nextEvents);
+  };
+
+  const handlePopoverOpen = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+    console.log('current target');
+  };
+
+  const showEventDoubleClick = (): void => {
+    //console.log('Event after double click');
```
[the remark this answers is not in the document]
Comment (hapalacios) [reply client/src/components/CalendarBoard/CalendarBoard.tsx]: Done. Thank you.

File: client/src/components/CalendarBoard/CalendarBoard.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const [events, setEvents] = useState({
+    start: moment().toDate(),
+    end: moment().add(1, 'days').toDate(),
+    title: 'Some title',0
+  });
+  */
+  const onEventResize = ({ start, end }: EventResizeProps): void => {
+    /*setEvents( ( {start: any, end: any }: any) => {
+      state.events[0].start = start;
+      state.events[0].end = end;
```
[the remark this answers is not in the document]
Comment (hapalacios) [reply client/src/components/CalendarBoard/CalendarBoard.tsx]: Done.

File: client/src/components/CalendarBoard/CalendarBoard.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import { DemoUserScheduleProps, DemoUserData } from './DemoUserEvent';
+
+const CalendarBoard = (): JSX.Element => {
+  const classes = useStyles();
+  //const [events, setEvent] = useState<IUserSchedule[]>(mockDatas);
+  const [events, setEvents] = useState<DemoUserScheduleProps[]>(DemoUserData);
+
+  /*
+  const [events, setEvents] = useState({
+    start: moment().toDate(),
```
[the remark this answers is not in the document]
Comment (hapalacios) [reply client/src/components/CalendarBoard/CalendarBoard.tsx]: Done.

File: client/src/components/CalendarBoard/CalendarBoard.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
+import React, { useState, useRef } from 'react';
+import { Grid } from '@material-ui/core';
+import { Calendar, momentLocalizer } from 'react-big-calendar';
+import moment from 'moment';
+import withDragAndDrop from 'react-big-calendar/lib/addons/dragAndDrop';
+
+import useStyles from './useStyles';
+
+import 'react-big-calendar/lib/addons/dragAndDrop/styles.css';
+//import 'react-big-calendar/lib/css/react-big-calendar.css'; // Default css by react-big-calendar dev team
```
[the remark this answers is not in the document]
Comment (hapalacios) [reply client/src/components/CalendarBoard/CalendarBoard.tsx]: Done.

File: client/src/components/ChatSideBanner/ChatSideBanner.tsx [hunk 6]
```diff
@@ -3,12 +3,12 @@ import Box from '@material-ui/core/Box';
...
 import Typography from '@material-ui/core/Typography';
 import useStyles from './useStyles';
-import { User } from '../../interface/User';
+//import { User } from '../../interface/User';
 import AvatarDisplay from '../AvatarDisplay/AvatarDisplay';
 import Search from '../Search/Search';
 import AuthMenu from '../AuthMenu/AuthMenu';
 
-interface Props {
+/*interface Props {
```
[the remark this answers is not in the document]
Comment (hapalacios) [reply client/src/components/ChatSideBanner/ChatSideBanner.tsx]: Sorry. This is a mistake. I should not make comments in this component at this moment.

File: client/src/pages/Dashboard/useStyles.ts [hunk 7]
```diff
@@ -8,6 +8,16 @@ const useStyles = makeStyles(() => ({
...
+  /*
+  drawerWrapper: {
+    width: drawerWidth,
+    [theme.breakpoints.up('md')]: {
+      width: '100%',
+      margin: 0,
+      padding: 0,
+      marginTop: '2.5%',
+    },
+  },*/
```
Comment (moffatethan) [inline client/src/pages/Dashboard/useStyles.ts:20]: Remove?

File: client/src/components/CalendarBoard/DemoUserEvent.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+const DemoUserData: DemoUserScheduleProps[] = [
+  DemoUserData1,
+  DemoUserData2,
+  DemoUserData3,
+  DemoUserData4,
+  DemoUserData5,
+  DemoUserData6,
+  DemoUserData7,
+];
+export { DemoUserData };
```
Comment (moffatethan) [inline client/src/components/CalendarBoard/DemoUserEvent.ts:59]: Good on this file for demo user, in the future the plan would be to have this data come from the back-end where we can recycle the data on a cron job every X hours?

**Review by moffatethan — APPROVED** [review]
Good work on this, but please let's take a look into the line sequencing issue more. It's causing troubles for reviewing and unnecessary large PRs. But overall, this was good work

**Pull request merged**